R/xp_utils.R

```diff
@@ -65,3 +65,17 @@ xp_find_location <- function(xml, xpath) {
     as.integer(xml_find_num(xml, xpath))
   }
 }
+
+#' Strip XPath 2.0-style comments from an XPath
+#'
+#' xml2 uses XPath 1.0, which has no support for comments. But comments are
+#'   useful in a codebase with as many XPaths as we maintain, so we fudge our
+#'   way to XPath 2.0-ish support by writing this simple function to remove comments.
+#'
+#' @noRd
+xpath_comment_re <- rex::rex(
+  "(:",
+  zero_or_more(not(":)")),
```

Comment:
What actual regex does this compile to?
Also XPath is not regular - comments are only allowed in "ignorable whitespace".
This approximation should be noted here:

`/STR_CONST[text() = '"(: not a comment :)"']` 

Would produce a wrong XPath.

Reply:
That's right -- but without a full XPath engine, we won't really be able to prevent that. And it seems unlikely to show up in anything we do in {lintr}. I'll make a note in the documentation.

> What actual regex does this compile to?

```regex
\(:(?:(?:(?!:\)).)*)*:\)
```

When I wrote it by hand I did `[(][:](?:[^:]|[:][^)])*[:][)]`, the simple `rex()` version is probably a little less efficient (by always checking two behind the current character), but easy to read.

Reply:
Thanks, both regexes seem acceptable.